### PR TITLE
add vertex prod features

### PR DESCRIPTION
**Pull request by baskaryan**
- chat vertex async
- vertex stream
- vertex full generation info
- vertex use server-side stopping
- model garden async
- update docs for all the above

in follow up will add
[] chat vertex full generation info
[] chat vertex retries
[] scheduled tests

### CODE REVIEW

**Comment by baskaryan** [comment]
cc @lkuligin

File: libs/langchain/langchain/chat_models/vertexai.py [hunk 1]
```diff
@@ -116,7 +122,11 @@ class ChatVertexAI(_VertexAICommon, BaseChatModel):
     """`Vertex AI` Chat large language models API."""
 
     model_name: str = "chat-bison"
-    streaming: bool = False
+    "Underlying model name."
+
+    @property
+    def _llm_type(self) -> str:
+        return "chat-vertexai"
```
Comment (baskaryan) [inline libs/langchain/langchain/chat_models/vertexai.py:129]: @dqbd will this cause problems

**Comment by ventz** [comment]
@baskaryan Not sure if related to the stop changes, but now getting:
(No issues pre .304 with the same code)

```
_TextGenerationModel.predict() got an unexpected keyword argument 'stop_sequences'
```

```
/langchain/llms/vertexai.py\", line 100, in _completion_with_retry\n    return llm.client.predict(*args, **kwargs)\nTypeError: _TextGenerationModel.predict() got an unexpected keyword argument 'stop_sequences
```

**Comment by lkuligin** [comment]
@ventz which version of `google-cloud-aiplatform` are you using? please, make sure you've updated to `1.31.1`  at least

**Comment by ventz** [comment]
@lkuligin Thanks -- just came back here to post exactly that :)

Hopefully this helps someone else!

**Pull request merged**